android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt

```diff
@@ -76,7 +76,9 @@ open class IpnViewModel : ViewModel() {
     viewModelScope.launch {
       Notifier.prefs.collect {
         it?.let { lastPrefs = it }
-        isRunningExitNode.set(it?.let { AdvertisedRoutesHelper.exitNodeOnFromPrefs(it) })
+        if (it != null) {
+          isRunningExitNode.set(AdvertisedRoutesHelper.exitNodeOnFromPrefs(it))
+        }
       }
```

Comment:
One could also write this whole block as

```
it?.let {
    lastPrefs = it
    isRunningExitNode.set(AdvertisedRoutesHelper.exitNodeOnFromPrefs(it))
}
```

Reply:
Also, do we need to set `isRunningExitNode` to `false` if there are no prefs?

Reply:
There are several places that check `isRunningExitNode`, so we could either have a default value or enforce that all checks do a null check